packages/driver/cypress/e2e/commands/waiting.cy.js

```diff
@@ -737,6 +737,44 @@ describe('src/cy/commands/waiting', () => {
       })
     })
 
+    describe('multiple alias execution order', () => {
+      beforeEach(() => {
+        cy.visit('/fixtures/jquery.html')
+      })
+
+      it('all responses returned in execution order', () => {
+        const resp1 = { value: 'alpha' }
+        const resp2 = { value: 'beta' }
+        const resp3 = { value: 'gamma' }
+        const resp4 = { value: 'delta' }
+        const resp5 = { value: 'epsilon' }
+
+        cy.intercept(/alpha/, resp1).as('getAlpha')
+        cy.intercept(/beta/, resp2).as('getBeta')
+        cy.intercept(/gamma/, resp3).as('getGamma')
+        cy.intercept(/delta/, resp4).as('getDelta')
+        cy.intercept(/epsilon/, resp5).as('getEpsilon')
+
+        cy.window().then((win) => {
+          xhrGet(win, '/epsilon')
+          xhrGet(win, '/beta')
+          xhrGet(win, '/gamma')
+          xhrGet(win, '/delta')
+          xhrGet(win, '/alpha')
+
+          return null
+        })
+
+        cy.wait(['@getAlpha', '@getBeta', '@getGamma', '@getDelta', '@getEpsilon']).then((responses) => {
```

Comment:
The [original issue](https://github.com/cypress-io/cypress/issues/27337) concerns using the same alias for a request that's called multiple times, but this is testing different aliases. Seems like this test should set up one alias on a particular request interception, trigger that request multiple times, and then assert that the responses are in the right order.

Reply:
Updated with new test.

Reply:
Now that I think about _this_ test some more ... is this actually the behavior we want? It seems counter-intuitive for the user to pass in the aliases in a specific order and then the responses are provided in a different order. I think the ordering may need to be more nuanced so it only affects aliases that are the same and not aliases that are different. So if the requests are:

```
alpha (1)
alpha (2)
beta
alpha (3)
gamma (1)
delta
gamma (2)
```

And the user's test has:

```js
cy.wait(['@getGamma', '@getBeta', '@getAlpha', '@getGamma', '@getDelta', '@getAlpha', '@getAlpha'])
```

The responses should be in the following order:

```js
.then((responses) => {
  responses[0] // => gamma (1)
  responses[1] // => beta
  responses[2] // => alpha (1)
  responses[3] // => gamma (2)
  responses[4] // => delta
  responses[5] // => alpha (2)
  responses[6] // => alpha (3)
})
```

Does that make sense?

Reply:
Yeah that makes sense, I'll take a crack at it. I wrote a version of that test and saw that behavior. Good Call!

Reply:
Updated to match.